pkg/solana/relay.go

```diff
@@ -162,7 +162,10 @@ func (r *Relayer) NewContractReader(_ context.Context, chainReaderConfig []byte)
 		return nil, fmt.Errorf("failed to init account reader err: %s", err)
 	}
 
-	return chainreader.NewChainReaderService(r.lggr, &chainreader.RPCClientWrapper{AccountReader: accountReader}, cfg)
+	// TODO: create a logpoller
+	var logpoller chainreader.EventsReader
```

Comment:
You don't need to create a new one here, it's already created in `NewChain()` which will have been called by the Relayer during its initialization:
https://github.com/smartcontractkit/chainlink-solana/blob/develop/pkg/solana/chain.go#L328-L329

All you should need is:
```suggestion
	logpoller = r.chain.LogPoller()
```
(or just pass that directly to `chainreader.NewContractReaderServices) You could validate that it's not nil I guess, either here or inside there

Reply:
One thing that hasn't been added yet though is lp to the list of subservices started in `chain.Start()`:
https://github.com/smartcontractkit/chainlink-solana/blob/develop/pkg/solana/chain.go#L524

I figure there's no reason to start it yet, since it's not used by anything in production at the moment (ie, by DataFeeds). Q; Are we planning on having a feature flag for CCIP? Or are we going to have ChainReader & LogPoller started regardless of whether it's just an ocr node or a ccip node?

Reply:
I just tried this and the interface doesn't have `FilteredLogs` which is necessary. Should we add this to the `LogPoller` interface expected by the relayer?

Reply:
Yes, I must have forgotten to add it when I added it to the `Service` struct in my latest PR--should be added